include/boost/multiprecision/gmp.hpp

```diff
@@ -2380,6 +2384,9 @@ struct gmp_rational
    template <class T, class U>
    gmp_rational(const T& a, const U& b, typename std::enable_if<std::is_constructible<gmp_int, T>::value && std::is_constructible<gmp_int, U>::value>::type* = nullptr)
    {
+      if (b == static_cast<U>(0))
```

Comment:
I don't think we have tests that would catch this, but I would worry about T and U being types which gmp_int is constructible from, but which do not have an `operator=(int)`, an example would be `mpz_t`.

So I would be inclined to put this check after the construction of the gmp_int's below and use a call to eval_is_zero for the check.  Hope that makes sense!

Reply:
I understand. This makes sense. i will tweak these and similar ones accordingly.